### PR TITLE
Feature/task 401 question component

### PR DESCRIPTION
**Pull request by MarcoGoC**
# Description 📝
New component to combine RadioGroup and Previous Next bar to display the questions 
Might need changes to get the value of the radio button clicked so far it is in the console 

## Taiga Link 🔗
https://taiga.dts-stn.com/project/ericdube-hfp-shared-backlog/task/401?kanban-status=134

## Checklist ✅
- [x] created story for all visual components
- [x] created or modified necessary unit tests
- [ ] created or modified e2e tests
- [ ] modified README or any applicable documentation on the changes you made

### CODE REVIEW

**Review by shewood — COMMENTED** [review]
I am just wondering about the skip button? Do we want to allow uses to skip some or all of the questions?

**Comment by MarcoGoC** [comment]
> I am just wondering about the skip button? Do we want to allow uses to skip some or all of the questions?

**Comment by MarcoGoC** [comment]
There is no skip button on your design :)

**Review by will0684 — APPROVED** [review]
Nice work Marco

File: src/components/molecules/Questions.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { RadioGroup } from "../molecules/RadioGroup";
+import { PrevNextBar } from "../molecules/PrevNextBar";
+
+export function Questions(props) {
+  const { id, required, textRequired, legend, name, options, onChange } = props;
+
+  function onChangeHandler(e) {
```
Comment (Moro-Code) [inline src/components/molecules/Questions.js]: whats the point of this extra function if its only passing in the event object.

File: src/components/molecules/PrevNextBar.js [hunk 2]
```diff
@@ -24,9 +25,9 @@ export function PrevNextBar(props) {
   }
 
   return (
-    <div id={id} className="flex p-2 inline-flex w-full">
+    <div id={id} className="flex p-2 bg-gray-300 justify-end">
       <Router>
```
Comment (Moro-Code) [inline src/components/molecules/PrevNextBar.js]: as in standup :)

File: src/components/molecules/Questions.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        id={id}
+        required={required}
+        textRequired={textRequired}
+        legend={legend}
+        name={name}
+        options={options}
+        onChange={() => onChangeHandler()}
+      />
+
+      <PrevNextBar
```
Comment (Moro-Code) [inline src/components/molecules/Questions.js]: when prev next is fixed pass handler for when next button or prev button is fixed

File: src/components/molecules/RadioGroup.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Legend } from "../atoms/Legend";
+import { Radio } from "../atoms/Radio";
+
+export function RadioGroup(props) {
```
Comment (Moro-Code) [inline src/components/molecules/RadioGroup.js:10]: just a brief description on what this component is

File: src/components/molecules/RadioGroup.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  const { id, required, textRequired, legend, name, options, onChange } = props;
+
+  function onChangeHandler(e) {
+    console.log(e);
```
Comment (Moro-Code) [inline src/components/molecules/RadioGroup.js]: make sure to remove all console logs in production code

File: src/components/molecules/RadioGroup.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Legend } from "../atoms/Legend";
+import { Radio } from "../atoms/Radio";
+
+export function RadioGroup(props) {
+  const { id, required, textRequired, legend, name, options, onChange } = props;
+
+  function onChangeHandler(e) {
```
Comment (Moro-Code) [inline src/components/molecules/RadioGroup.js]: not a useful function

**Review by Moro-Code — DISMISSED** [review]
thanks marco ! Great work

File: src/components/molecules/PrevNextBar.js (hunk 2 above)
Comment (MarcoGoC) [reply src/components/molecules/PrevNextBar.js]: created a PrevNext component instead . Thanks

File: src/components/molecules/PrevNext.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
+import React from "react";
```
Comment (Moro-Code) [inline src/components/molecules/PrevNext.js:1]: I can't seem to see the story for this component in the storybook UI. Can you validate ?

File: src/components/molecules/PrevNextBar.js (hunk 2 above)
Comment (Moro-Code) [reply src/components/molecules/PrevNextBar.js]: Can you delete this component if its not being used

File: src/components/molecules/Questions.stories.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+  options: [
+    { id: "0", value: "A", label: "A) Less than $30,000" },
+    { id: "1", value: "B", label: "B) Between $30,000 & 60,000" },
+    { id: "2", value: "C", label: "C) More than $60,000" },
+  ],
+  onChange: (e) => onChangeHandler(e),
+  onPrevClick: () => onPrevHandler(),
+  prevText: "Previous Question",
+  nextText: "Next Question",
+  onNextClick: () => onNextHandler(),
```
Comment (Moro-Code) [inline src/components/molecules/Questions.stories.js]: remove this, storybooks automatically passes functions for callbacks like this

File: src/components/molecules/Questions.stories.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+  id: "Question1",
+  required: true,
+  textRequired: "(required)",
+  options: [
+    { id: "0", value: "A", label: "A) Less than $30,000" },
+    { id: "1", value: "B", label: "B) Between $30,000 & 60,000" },
+    { id: "2", value: "C", label: "C) More than $60,000" },
+  ],
+  onChange: (e) => onChangeHandler(e),
+  onPrevClick: () => onPrevHandler(),
```
Comment (Moro-Code) [inline src/components/molecules/Questions.stories.js]: remove

File: src/components/molecules/Questions.stories.js [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+  name: "Q1",
+  id: "Question1",
+  required: true,
+  textRequired: "(required)",
+  options: [
+    { id: "0", value: "A", label: "A) Less than $30,000" },
+    { id: "1", value: "B", label: "B) Between $30,000 & 60,000" },
+    { id: "2", value: "C", label: "C) More than $60,000" },
+  ],
+  onChange: (e) => onChangeHandler(e),
```
Comment (Moro-Code) [inline src/components/molecules/Questions.stories.js:29]: remove

**Review by Moro-Code — DISMISSED** [review]
minor changes needed thanks Marco

File: src/components/molecules/Questions.stories.js (hunk 8 above)
Comment (MarcoGoC) [reply src/components/molecules/Questions.stories.js]: Thanks for the review.

File: src/components/molecules/Questions.stories.js (hunk 10 above)
Comment (MarcoGoC) [reply src/components/molecules/Questions.stories.js:29]: I have to put it back as without it the test fails. 
did something similar to the header component for onLanguageClick just to pass the test

Comment (Moro-Code) [reply src/components/molecules/Questions.stories.js:29]: you could have passed a mock function using jest.fn

**Pull request merged**